packages/eslint-plugin/src/rules/await-thenable.ts

```diff
@@ -79,11 +79,18 @@ export default createRule<[], MessageId>({
           return;
         }
 
-        const asyncIteratorSymbol = tsutils.getWellKnownSymbolPropertyOfType(
-          type,
-          'asyncIterator',
-          checker,
-        );
+        const types = type.isUnion() ? type.types : [type];
+
+        const asyncIteratorSymbol =
+          types
+            .map(t =>
+              tsutils.getWellKnownSymbolPropertyOfType(
+                t,
+                'asyncIterator',
+                checker,
+              ),
+            )
+            .find(symbol => symbol) ?? null;
```

Comment:
(nit)
```suggestion
            .find(symbol => symbol != null);
```

Reply:
I need to update the condition at line 95 with the following:
```ts
if (asyncIteratorSymbol === undefined) {
```
is it ok ?

Reply:
done

Reply:
Good to know, thanks !
I will revert my change to bring back `== null`